src/components/modules/toolbar/inline.ts

```diff
@@ -362,20 +378,76 @@ export default class InlineToolbar extends Module {
     }
 
     const toolSettings = this.Editor.Tools.getToolSettings(currentBlock.name);
+    const inlineToolbarSettingsForTool = toolSettings && toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
 
-    return toolSettings && toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
+    /**
+     * Condition 1 : Returns true, if the inlineToolbar property of the block is an array or true;
+     * false otherwise.
+     */
+    const isInlineToolbarSettingsForToolProvided = inlineToolbarSettingsForTool == true || Array.isArray(inlineToolbarSettingsForTool);
+
+    /**
+     * Condition 2 : Returns true, if the inlineToolbar property of the block is not set,
+     * but the default inlineToolbar property is either an array or true.
+     */
+    const isDefaultInlineToolbarProvided = inlineToolbarSettingsForTool == undefined && this.config.inlineToolbar;
+
+    /**
+     * If one of the above condition is true, then the inlineToolbar will be displayed.
+     */
+    const showInlineToolbar = isInlineToolbarSettingsForToolProvided || isDefaultInlineToolbarProvided;
+
+    /**
+     * condition 2 could return an array, so casting it to Boolean.
+     */
+    return Boolean(showInlineToolbar);
+  }
+
+  /**
+   * Sort tools according to the given order in inlineToolbar property
+   *
+   * @param inlineToolbar - inlineToolbar order.
+   */
+  private sortTools(inlineToolbar: string[]): HTMLElement[] {
+    /**
+     * Gets all the button nodes with class `ce-inline-tool`
+     */
+    const buttonNodes = this.nodes.buttons.querySelectorAll(`.${this.CSS.inlineToolButton}`);
+
+    /**
+     * creates an array of all the button nodes
+     */
+    const buttons = Array.from(buttonNodes);
+
+    /**
+     * sorts the buttons array according to the inlineToolbar property provided by user
+     */
+    inlineToolbar.forEach((toolName, i) => {
```

Comment:
Can't we use builtin `sort` method here?

Reply:
We have to sort it according to the given array. Is there a built in sort method present for this?

Reply:
@gohabereg answer pls

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort